solo-io/gloo
# test/kube/e2e: introduce testify, create example suite

# Description

Introduce an example test suite, which relies on testify, and demonstrates how we can re-use an existing testing framework to remove a lot of the code that our custom framework introduced.


# Context

Adds testify:  https://github.com/stretchr/testify/tree/master

Introduces an example suite, not run in CI, to demonstrate how to group and execute our tests

**Follow-Up Work**:
- Merge this PR, once there is agreement
- Convert existing tests to this structure, and get them passing in CI
- Write enterprise test suites to demonstrate executing these tests

## Interesting decisions
 -

## Testing steps
This example suite does not run in CI. However, I changed some code that is run by our other suites, and those run in CI. So we expect those to pass.

If you want to run the example suite, try out the following:

```
kind create cluster
```

```
RELEASED_VERSION=1.17.0-beta20 CLUSTER_NAME=kind go test ./test/kubernetes/e2e/example/ -v
```

## Notes for reviewers

Be sure to verify intended behavior by ...

Please proofread comments on ...

This is a complex PR and may require a huddle to discuss ...

# Checklist:
